### PR TITLE
Keras examples for single input and multiple inputs have been added.

### PR DESCRIPTION
**Pull request by justnoxx**
These examples are explaining how to use saved models created using Keras.
1st example explains how to use the model when there is precisely one input operation.
2nd example shows how to deal with models where more than one input ops.

### CODE REVIEW

**Comment by justnoxx** [comment]
@googlebot I signed it!

**Comment by justnoxx** [comment]
@adamcrume there are some examples related to Keras that I mentioned in https://github.com/tensorflow/rust/issues/282
Please, let me know what I should change if required to make it merged.
Thanks.

/Dmitriy

File: examples/keras_multiple_inputs_saved_model/keras_multiple_inputs_saved_model.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+model = Model(inputs=[input1, input2], outputs=dense_layer)
+
+v1 = np.array([[0.1, 0.2, 0.3, 0.4, 0.5]])
+v2 = np.array([[0.6, 0.7, 0.8, 0.9, 0.1]])
+print(v1.shape)
+x1, x2 = np.random.randn(100, 5), np.random.randn(100, 5)
+print(x1.shape)
+y = np.random.randn(100, 1)
+# print(v1.shape)
+# print(v2.shape)
```
Comment (adamcrume) [inline examples/keras_multiple_inputs_saved_model/keras_multiple_inputs_saved_model.py:35]: Were these commented-out lines intended to be removed?

File: examples/keras_signle_input_saved_model.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+    let out_res:f32 = args.fetch(out).unwrap()[0];
+    println!("Results: {:?}", out_res);
+}
+
+pub fn vector_to_tensor(v: &Vec<f32>) -> Tensor<f32> {
+    let dimension = v.len();
+    let tensor = Tensor::new(&[1, dimension as u64]).with_values(&v[..]).unwrap();
+    return tensor;
+}
```
Comment (adamcrume) [inline examples/keras_signle_input_saved_model.rs:45]: This file name has a typo: signle should be single.

File: examples/keras_signle_input_saved_model.rs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+fn main() {
+    // In this file test_in_input is being used while in the python script,
+    // that generates the saved model from Keras model it has a name "test_in".
+    // For multiple inputs _input is not being appended to the op name.
+    let input_op_name = "test_in_input";
+    let output_op_name = "test_out";
+    let save_dir = "examples/keras_signle_input_saved_model";
+    let v: Vec<f32> = vec![0.1, 0.2, 0.3, 0.4, 0.5];
+
+    let tensor = vector_to_tensor(&v);
```
Comment (adamcrume) [inline examples/keras_signle_input_saved_model.rs:12]: As with the other file, this can be simplified to:

```
let tensor = Tensor::from(&[0.1, 0.2, 0.3, 0.4, 0.5][..]);
```

File: examples/keras_signle_input_saved_model.rs [hunk 4]
```diff
@@ -0,0 +1,45 @@
+use tensorflow::{SessionRunArgs, Graph, SessionOptions, Tensor, SavedModelBundle};
+
+fn main() {
+    // In this file test_in_input is being used while in the python script,
+    // that generates the saved model from Keras model it has a name "test_in".
+    // For multiple inputs _input is not being appended to the op name.
+    let input_op_name = "test_in_input";
```
Comment (adamcrume) [inline examples/keras_signle_input_saved_model.rs:7]: These aren't op names.

File: examples/keras_signle_input_saved_model.rs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    let input_op = graph.operation_by_name_required(&input_info.name().name).unwrap();
+    let output_op = graph.operation_by_name_required(&output_info.name().name).unwrap();
+    let mut args = SessionRunArgs::new();
+    args.add_feed(&input_op, 0, &tensor);
+    let out = args.request_fetch(&output_op, 0);
+
+    let result = session.run(&mut args);
+    if result.is_err() {
+        panic!("Error occured during calculations: {:?}", result);
+    }
```
Comment (adamcrume) [inline examples/keras_signle_input_saved_model.rs:36]: This can use `expect`.

**Review by adamcrume — CHANGES_REQUESTED** [review]
Looks good, just a few minor style issues.

Also, please run `cargo fmt`.

File: examples/keras_multiple_inputs_saved_model.rs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    // In this file test_in_input is being used while in the python script,
+    // that generates the saved model from Keras model it has a name "test_in".
+    // For multiple inputs _input is not being appended to the op name.
+    let input_1_op_name = "test_in1";
+    let input_2_op_name = "test_in2";
+    let output_op_name = "test_out";
+    let save_dir = "examples/keras_multiple_inputs_saved_model";
+    let v1: Vec<f32> = vec![0.1, 0.2, 0.3, 0.4, 0.5];
+    let v2 = vec![0.6, 0.7, 0.8, 0.9, 0.1];
+    let tensor1 = vector_to_tensor(&v1);
```
[the remark this answers is not in the document]
Comment (justnoxx) [reply examples/keras_multiple_inputs_saved_model.rs]: Function is removed.

File: examples/keras_multiple_inputs_saved_model.rs [hunk 7]
```diff
@@ -0,0 +1,51 @@
+use tensorflow::{SessionRunArgs, Graph, SessionOptions, Tensor, SavedModelBundle};
+
+fn main() {
+    // In this file test_in_input is being used while in the python script,
+    // that generates the saved model from Keras model it has a name "test_in".
+    // For multiple inputs _input is not being appended to the op name.
+    let input_1_op_name = "test_in1";
```
[the remark this answers is not in the document]
Comment (justnoxx) [reply examples/keras_multiple_inputs_saved_model.rs]: Variable is renamed now.

File: examples/keras_multiple_inputs_saved_model.rs [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    let output_op = graph.operation_by_name_required(&output_info.name().name).unwrap();
+    let mut args = SessionRunArgs::new();
+    args.add_feed(&input_op1, 0, &tensor1);
+    args.add_feed(&input_op2, 0, &tensor2);
+    let out = args.request_fetch(&output_op, 0);
+
+    let result = session.run(&mut args);
+    if result.is_err() {
+        panic!("Error occured during calculations: {:?}", result);
+    }
```
[the remark this answers is not in the document]
Comment (justnoxx) [reply examples/keras_multiple_inputs_saved_model.rs]: Simplification has been performed.

File: examples/keras_multiple_inputs_saved_model/keras_multiple_inputs_saved_model.py (hunk 1 above)
Comment (justnoxx) [reply examples/keras_multiple_inputs_saved_model/keras_multiple_inputs_saved_model.py]: Obsolete lines have been removed.

File: examples/keras_signle_input_saved_model.rs (hunk 2 above)
Comment (justnoxx) [reply examples/keras_signle_input_saved_model.rs]: File has been renamed

File: examples/keras_signle_input_saved_model.rs (hunk 3 above)
Comment (justnoxx) [reply examples/keras_signle_input_saved_model.rs]: Well, it did not work for me because your code is creating tensor of dims:[5], but dim:[1, 5] is expected, so I used the following code:

```rust
let tensor: Tensor<f32> = Tensor::new(&[1, 5])
        .with_values(&[0.1, 0.2, 0.3, 0.4, 0.5])
        .expect("Can't create tensor");
```

File: examples/keras_signle_input_saved_model.rs (hunk 4 above)
Comment (justnoxx) [reply examples/keras_signle_input_saved_model.rs]: Variable has been renamed.

File: examples/keras_signle_input_saved_model.rs (hunk 5 above)
Comment (justnoxx) [reply examples/keras_signle_input_saved_model.rs]: Expect has been added.

**Comment by justnoxx** [comment]
@adamcrume sorry for delay, literally did not have time to address that, but now I did the changes that you requested. A file with typo in its name has been renamed. Cargo fmt has been also applied. Please, let me know if I need to do additional updates.
